File: day:19/deleteDuplicates.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        #base case
        if(head is None): return head;
        if(head.next is None): return head;
    
        current = head;
    
        while(True):
            next = current.next;
            if(next is None): break;
            while(next and (next.val == current.val)):
                next = next.next;
            if(next):
                current.next = next;
                current = next;
            else:
                current.next = None;
                break;
    
        return head;

File: day:19/deleteDuplicates.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        // base case
        if(head == null) return head;
        if(head.next == null) return head;
    
        var current = head;
    
        while(true) {
            var next = current.next;
            if(next == null) break;
            while( (next != null) && (next.val == current.val)) {
                next = next.next;
            }
            if(next != null) {
                current.next = next;
                current = next;
            }
            else {
                current.next = null;
                break;
            }
        }
    
        return head;
        
        }
}
